README.md

```diff
@@ -179,19 +272,43 @@ For example, the following YAML portion will expose two dynamic commands, `drupa
 
 ```yaml
 commands:
+  # The name of the command, in the format "group:commandname".
   drupal:site-setup:
+    # A list of tasks that will be executed in sequence. Any value using shell
+    # variable notation "${config.option}" will be replaced by the config value.
     - { task: "chmod", file: "${drupal.root}/sites", permissions: 0774, recursive: true }
     - { task: "symlink", from: "../../custom/modules", to: "${drupal.root}/modules/custom" }
     - { task: "symlink", from: "../../custom/themes", to: "${drupal.root}/themes/custom" }
+    # Commands can reference each other using the "run" task, allowing for
+    # complex scenarios to be implemented with relative ease.
     - { task: "run", command: "drupal:drush-setup" }
     - { task: "run", command: "drupal:settings-setup" }
     - { task: "run", command: "setup:behat" }
+    # A shell command can be executed by including it as a string. For more
+    # advanced cases it is recommended to use the "exec" task instead.
     - "./vendor/bin/drush --root=$(pwd)/${drupal.root} cr"
-  setup:behat:
-    - { task: "process", source: "behat.yml.dist", destination: "behat.yml" }
+  deploy:rsync-files:
+    - task: exec
+      command: rsync
+      # Define the command arguments as an array.
+      arguments:
+        - ${files.source_path}
+        - ${files.destination_path}
+      options:
+        # Options that have the value `null` are passed without a value, this
+        # will result in `--archive --compress --verbose`.
+        archive: null
+        compress: null
+        verbose: null
+        # Any specified values will be passed to the option: `--info=del`.
+        info: del
+        # Some options can be passed multiple times. In this case define the
+        # values as an array. This outputs `--exclude=cache --exclude=vendor`.
+        exclude:
+          - cache
+          - vendor
```

Comment:
I added the documentation of the feature added here, and also gave the README a much needed overhaul.

Reply:
I'm not sure, the config parte has multiple problems that cannot be fixed in 1.x due to potential BC break. I found at least 2: #138 & #135